samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -304,6 +307,17 @@ private void loadMetadataResources(JobModel jobModel) {
         byte[] serializedValue = jsonSerde.toBytes(entry.getValue());
         configStore.put(entry.getKey(), serializedValue);
       }
+
+      // fan out the startpoints
+      StartpointManager startpointManager = createStartpointManager(coordinatorStreamStore);
+      startpointManager.start();
+      try {
+        startpointManager.fanOut(JobModelUtil.getTaskToSystemStreamPartitions(jobModel));
+      } finally {
+        startpointManager.stop();
+      }
+    } catch (IOException ex) {
```

Comment:
Minor: Why do we need to wrap the exception and throw-back. Wouldn't bluntly throwing back here would suffice?

Reply:
Just to wrap it as a `RuntimeException` which `SamzaException` extends so the exception will propagate up as an unchecked exception and fail the job.